code/game/machinery/Sleeper.dm

```diff
@@ -59,18 +98,113 @@
+	if(istype(I, /obj/item/reagent_containers))
+		var/chem_stores_len = LAZYLEN(chem_stores)
+		if(!chem_stores_len)
+			to_chat(user, "<span class='warning'>[src] doesn't have any chemical storage slots.</span>")
+		else if(!selected_store || chem_stores_len < selected_store)
+			to_chat(user, "<span class='warning'>There doesn't seem to be any where to put this, select a slot first!</span>")
+		else if(chem_stores[selected_store])
+			to_chat(user, "<span class='warning'>There's already a container in this slot, eject it first or select another one!</span>")
+		else
+			if(user.transferItemToLoc(I, src))
+				add_store(I)
+		return TRUE
+
+/obj/machinery/sleeper/proc/add_store(obj/item/reagent_containers/C, store_index)
+	store_index = store_index || selected_store
+	if(store_index && LAZYLEN(chem_stores) >= store_index)
+		if(!chem_stores[store_index])
+			chem_stores[store_index] = C
+
+/obj/machinery/sleeper/proc/eject_store(store_index)
+	var/obj/item/reagent_containers/C = LAZYACCESS(chem_stores, store_index)
+	if(C)
+		var/atom/L = drop_location()
+		if(L)
+			C.forceMove(L)
+		chem_stores[store_index] = null //if L is null then the sleeper is probably in nullspace so we still want to null this ref
+		return C
+
+/obj/machinery/sleeper/proc/inject_store(store_index)
+	var/obj/item/reagent_containers/C = LAZYACCESS(chem_stores, store_index)
+	var/datum/reagents/R = C.reagents
+	if(C && R && R.total_volume)
+		var/mob/living/mob_occupant = occupant
+		if(!mob_occupant.is_injectable())
+			return
+		var/coeff = min(C.amount_per_transfer_from_this/R.total_volume, 1)
+		R.reaction(mob_occupant, INJECT, coeff)
+		R.trans_to(mob_occupant, C.amount_per_transfer_from_this)
+
+/obj/machinery/sleeper/proc/name_store(store_index, name)
+	var/chem_stores_len = LAZYLEN(chem_stores)
+	if(chem_stores_len && chem_stores_len >= store_index)
+		name = sanitize(trim(name, 21))
+		var/chem_store_names_len = LAZYLEN(chem_store_names)
+		if(!chem_store_names || chem_store_names_len < chem_stores_len)
+			LAZYINITLIST(chem_store_names)
+			chem_store_names.len = chem_stores_len
+		chem_store_names[store_index] = name
+
+/obj/machinery/sleeper/proc/chill_out(mob/living/target)
+	var/freq = rand(24750, 26550)
+	playsound(src, 'sound/effects/spray.ogg', 5, TRUE, 2, frequency = freq)
+	target.SetStasis(TRUE)
+	target.remote_control = src
+	target.ExtinguishMob()
+
+/obj/machinery/sleeper/proc/set_stasis(enabled)
+	stasis_enabled = enabled
+	var/mob/living/mob_occupant = occupant
+	if(!mob_occupant)
+		return
+	if(enabled)
+		chill_out(mob_occupant)
+	else
+		mob_occupant.SetStasis(FALSE)
+
+/obj/machinery/sleeper/close_machine(mob/living/user)
 	if((isnull(user) || istype(user)) && state_open && !panel_open)
 		..(user)
 		var/mob/living/mob_occupant = occupant
-		if(mob_occupant && mob_occupant.stat != DEAD)
-			to_chat(occupant, "[enter_message]")
+		if(mob_occupant)
+			if(stasis_enabled)
+				chill_out(mob_occupant)
+			mob_occupant.remote_control = src

```

Comment:
Duplicate code handled in chill_out

On another note, should this use buckling instead? I don't think anything currently uses buckling while putting the mob inside itself.